source/frontend/src/features/properties/map/project/add/AddProjectFileYupSchema.ts

```diff
@@ -9,4 +10,33 @@ export const AddProjectYupSchema = Yup.object().shape({
   projectStatusType: Yup.string(),
   region: Yup.string().required('Region is required.'),
   summary: Yup.string().max(2000, 'Project summary must be at most ${max} characters'),
+  products: Yup.array().of(
+    Yup.object().shape({
+      code: Yup.string()
+        .required('Product Code is required')
+        .max(20, 'Product code must be at most ${max} characters'),
+      description: Yup.string()
+        .required('Product Description is required')
+        .max(200, 'Product description must be at most ${max} characters'),
+      costEstimate: Yup.lazy(value =>
+        value === ''
+          ? Yup.string()
+          : Yup.number().typeError('Cost estimate must be a number').max(MAX_SQL_MONEY_SIZE),
+      ),
+      startDate: Yup.date(),
+      /*costEstimateDate: Yup.date().when('costEstimate', (costEstimate: string, schema: any) =>
```

Comment:
hmm, what's the plan with these commented out validations? are these for a future story or something?

Reply:
There is a discrepancy between confluence and the UX/UI page on the estimatedate being requried. I will remove the comments.